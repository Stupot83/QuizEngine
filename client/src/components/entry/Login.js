import React, { Component } from "react";
import { Link } from "react-router-dom";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import { loginUser } from "../../actions/actionsForAuthentication";
import "../../sass/Login.scss";
import Button from "@material-ui/core/Button";
import LockOpenIcon from "@material-ui/icons/LockOpen";

class Login extends Component {
    constructor() {
        super();
        this.state = {
            email: "",
            password: "",
            invalidEntries: {}
        };
    }

    componentDidMount() {
        // Redirect User to home page if already authenticated
        if (this.props.auth.isAuthenticated) {
            this.props.history.push("/home");
        }
    }

    componentWillReceiveProps(nextProps) {
        if (nextProps.auth.isAuthenticated) {
            this.props.history.push("/home"); // Redirect User to Home Page once logged in
        }
        if (nextProps.invalidEntries) {
            this.setState({
                invalidEntries: nextProps.invalidEntries
            });
        }
    }

    onChange = e => {
        this.setState({ [e.target.id]: e.target.value });
    };

    onSubmit = e => {
        e.preventDefault();

        const userData = {
            email: this.state.email,
            password: this.state.password
        };

        this.props.loginUser(userData);
    };

    render() {
        const { invalidEntries } = this.state;

        return (
            <div className="Login_container">
                <div id="Login_header" className="Login_header">
                    Login to QuizEngine
                </div>

                <form className="Login_form" noValidate onSubmit={this.onSubmit}>
                    <div className="Login_details">
                        <label>
                            <div id="Email_header" className="Login_label">
                                Email
                            </div>
                            <input
                                onChange={this.onChange}
                                value={this.state.email}
                                error={invalidEntries.email}
                                id="email"
                                type="email"
                                autoComplete="off"
                                className="Login_field"
                            />
                            <div className="Login-invalid_entries">{invalidEntries.email}</div>
                        </label>
                    </div>

                    <div className="Login_details">
                        <label>
                            <div id="Password_header" className="Login_label">
                                Password
                            </div>
                            <input
                                onChange={this.onChange}
                                value={this.state.password}
                                error={invalidEntries.password}
                                id="password"
                                type="password"
                                autoComplete="off"
                                className="Login_field"
                            />
                            <div className="Login_invalid_entries">{invalidEntries.password}</div>
                        </label>
                    </div>

                    <div className="Login_button_box">
                        <Button
                            id="Login_button"
                            type="submit"
                            variant="contained"
                            size="large"
                            className="Login_button"
                            startIcon={<LockOpenIcon />}
                        >
                            Login
                        </Button>
                    </div>
                    <div className="Login_bottom">
                        <p id="Redirect_message">
                            Not registered for QuizEngine?{" "}
                            <Link to="/register" id="Registration_link" className="Login_registration_link">
                                Register
                            </Link>
                        </p>
                    </div>
                </form>
            </div>
        );
    }
}

Login.propTypes = {
    loginUser: PropTypes.func.isRequired,
    auth: PropTypes.object.isRequired,
    invalidEntries: PropTypes.object.isRequired,
};

const mapStateToProps = state => ({
    auth: state.auth,
    invalidEntries: state.invalidEntries
});

export default connect(mapStateToProps, { loginUser })(Login);
