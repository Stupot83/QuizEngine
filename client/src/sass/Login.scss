.Login_container {
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    height: 100vh;
    background-image: url("https://www.transparenttextures.com/patterns/brick-wall.png");

    .Login_header {
        color: #3454d1;
        padding: 2rem;
        font-size: 20px;
        text-align: center;
        font-weight: bold;
        margin-bottom: 20px;
    }

    .Login_form {
        width: 500px;
        background-color: ghostwhite;
        border: 1px solid ghostwhite;
        padding: 2rem;
    }

    .Login_details {
        margin-bottom: 1rem;

        .Login_label {
            width: 100%;
            padding-right: 2rem;
            margin-bottom: 0.75rem;
            font-size: 14px;
        }

        .Login_field {
            background-color: white;
            border-bottom: 1px solid #3454d1;
            height: 40px;
            width: 90%;
            line-height: 1.5em;
            outline: 0;
            transition: all 0.25s ease 0s;
            font-size: 16px;
            padding: 0 1rem;

            &:hover {
                transition: all 0.25s ease 0s;
                background-color: #efefef;
            }

            &:focus {
                transition: all 0.25s ease 0s;
                background-color: #efefef;
            }
        }

        .Login_invalid_entries {
            margin-top: 0.5rem;
            color: red;
            font-size: 14px;
        }
    }

    .Login_button_box {
        text-align: center;
        margin: 40px;

        .Login_button {
            min-width: 150px;
            margin-left: 30px;
            margin-right: 30px;
            padding: 15px;
            border-radius: 15px;
            background-color: #3454d1;
            color: ghostwhite;
            font-weight: bold;
        }
    }

    .Login_bottom {
        display: flex;
        justify-content: space-around;
        margin-top: 2rem;

        .Login_registration_link {
            text-decoration: none !important;
            color: #3454d1;
        }
    }

    @media (max-width: 800px) {
        .Login_bottom {
            flex-direction: column;
            padding-left: 0px;
            text-align: center;
        }

        .Login_form {
            width: 80%;
        }
    }

    @media (max-width: 425px) {
        .Login_form {
          width: 300px !important;
        }
      }
}
