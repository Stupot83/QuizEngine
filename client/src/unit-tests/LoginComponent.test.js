import React from "react";
import { shallow } from "enzyme";
import Login from "../components/entry/Login";
import QuizEngineStore from "../quizEngineStore";
import { Provider } from "react-redux";

const mockStore = QuizEngineStore;

describe("Loading Component", () => {
    it("should render correctly in debug mode", () => {
        const component = shallow(
            <Provider store={mockStore}>
                <Login debug />
            </Provider>
        );

        expect(component).toMatchSnapshot();
    });

    it("should render correctly", () => {
        const component = shallow(
            <Provider store={mockStore}>
                <Login />
            </Provider>
        );

        expect(component).toMatchSnapshot();
    });
});
