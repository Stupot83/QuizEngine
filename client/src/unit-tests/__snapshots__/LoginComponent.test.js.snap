// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading Component should render correctly 1`] = `ShallowWrapper {}`;

exports[`Loading Component should render correctly in debug mode 1`] = `ShallowWrapper {}`;
